Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -477,6 +477,56 @@ void TextureCacheBase::BindTextures()
   }
 }
 
+struct ManualMipmapDetector
+{
+
+  ManualMipmapDetector()
+  {
+    min = { 255, 255, 255, 255 };
+    max = { 0, 0, 0, 0 };
+  }
+
+  void AddLevel(u32 width, u32 height, u32 row_length, const u8* buffer)
+  {
+    ++levels;
+
+    u64 avg[4] = { 0 };
+    for (u32 i = 0; i < height; ++i)
+    {
+      auto* row = buffer;
+      for (u32 j = 0; j < width; ++j)
+      {
+        avg[0] += row[0];
+        avg[1] += row[1];
+        avg[2] += row[2];
+        avg[3] += row[3];
+
+        row += 4;
```

Comment:
```cpp
row += avg.size();
```

Reply:
actually the row increment here is hardcoded in the incoming data, so I think that's a false abstraction. You aren't free to change `avg.size()` without breaking the code like that, so it actually is more fragile I think.